src/jitlayers.cpp

```diff
@@ -78,16 +78,21 @@ void jl_jit_globals(std::map<void *, GlobalVariable*> &globals)
 extern "C" JL_DLLEXPORT
 uint64_t jl_cumulative_compile_time_ns_before()
 {
-    int tid = jl_threadid();
-    jl_measure_compile_time[tid] = 1;
-    return jl_cumulative_compile_time[tid];
+    // Increment the flag to allow reentrant callers to `@time`.
+    jl_atomic_fetch_add(&jl_measure_compile_time, 1);
+    return jl_atomic_load(&jl_cumulative_compile_time);
```

Comment:
```suggestion
    return jl_atomic_fetch_add(&jl_measure_compile_time, 1) + 1;
```

(as with my first comment here, the fetch much only happen once, or it is not atomic, merely data-race-free)

Reply:
EDIT: oh, I see those are different variables. I suppose I am not sure how this part is intended to work, but I think your change is as close to correct as would currently be possible.

Reply:
I think the alternative would be having a shared global lock for both of those variables, but i don't think it really adds much to this function, since if compilation is going on elsewhere, it's already a bit arbitrary whether or not you return that compilation time.

I agree, i think this current implementation seems okay, and I'll probably leave this one alone?

Also - i mixed up those two variable names multiple times too. 😑  Maybe I should rename them to indicate that one is a boolean/counter? Something like `jl_measure_compile_time_enabled`?

Reply:
Done - i've renamed it to jl_measure_compile_time_enabled (and fixed another mix-up i caught along the way 😅)